bulkExport/glueScripts/export-script.py

```diff
@@ -0,0 +1,116 @@
+from awsglue.context import GlueContext
+from awsglue.job import Job
+from awsglue.dynamicframe import DynamicFrame
+from datetime import datetime
+
+glueContext = GlueContext(SparkContext.getOrCreate())
+job = Job(glueContext)
+
+args = getResolvedOptions(sys.argv, ['JOB_NAME', 'jobId', 'exportType', 'transactionTime', 'since', 'outputFormat', 'glueDatabase', 'glueTableName', 's3OutputBucket'])
+
+# type and groupId are optional parameters
+type = None
+if ('--{}'.format('type') in sys.argv):
+    type = getResolvedOptions(sys.argv, ['type'])['type']
+groupId = None
+if ('--{}'.format('groupId') in sys.argv):
+    groupId = getResolvedOptions(sys.argv, ['groupId'])['groupId']
+
+job_id = args['jobId']
+export_type = args['exportType']
+transaction_time = args['transactionTime']
+since = args['since']
+outputFormat = args['outputFormat']
+glue_table_name = args['glueTableName']
+
+glue_database = args['glueDatabase']
+bucket_name = args['s3OutputBucket']
+
+# Read data from DDB
+original_data_source_dyn_frame = glueContext.create_dynamic_frame.from_catalog(database = glue_database, table_name = glue_table_name)
+
+print('Start filtering by transactionTime and Since')
+# Filter by transactionTime and Since
+datetime_since = datetime.strptime(since, "%Y-%m-%dT%H:%M:%S.%fZ")
+datetime_transaction_time = datetime.strptime(transaction_time, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+filtered_dates_dyn_frame = Filter.apply(frame = original_data_source_dyn_frame,
+                           f = lambda x:
+                           datetime.strptime(x["meta"]["lastUpdated"], "%Y-%m-%dT%H:%M:%S.%fZ") > datetime_since and
+                           datetime.strptime(x["meta"]["lastUpdated"], "%Y-%m-%dT%H:%M:%S.%fZ") <= datetime_transaction_time
+                          )
+
+print('Start filtering by documentStatus and resourceType')
+# Filter by resource listed in Type and with correct STATUS
+type_list = None if type == None else type.split(',')
+valid_document_state_to_be_read_from = ['AVAILABLE','LOCKED', 'PENDING_DELETE']
+filtered_dates_resource_dyn_frame = Filter.apply(frame = filtered_dates_dyn_frame,
+                                    f = lambda x:
+                                    f = lambda x:
+                                    x["documentStatus"] in valid_document_state_to_be_read_from if type_list is None
+                                    else x["documentStatus"] in valid_document_state_to_be_read_from and x["resourceType"] in type_list
+                          )
+
+if filtered_dates_resource_dyn_frame.count() > 0:
+    print('Dropping fields not needed')
+    # Drop fields that are not needed
+    data_source_cleaned_dyn_frame = DropFields.apply(frame = filtered_dates_resource_dyn_frame, paths = ['documentStatus', 'lockEndTs', 'vid'])
+
+    print('Combining data into smaller partitions')
+    # Combine data into at most 10 partitions
+    data_frame = data_source_cleaned_dyn_frame.toDF()
+    data_frame = data_frame.coalesce(10)
```

Comment:
Unfortunately, combining all of the data into 1 file is very slow and cause the Glue job to fail with an out of memory exception. Splitting the data into at most 10 files per `resourceType` allow the Glue job to export 1 GB of data in 10 minutes.

Reply:
why 10? can it be a static number?
I feel that it's going to fail with a similar out of memory error for for a DB that is 10x the size. But idk for sure

Reply:
I did some further testing, and with G2x worker type, Glue can partition 1GB of data into 1 file per resource type. To keep things simple, I'm planning on setting a hard limit. When we have less than 100 MB of data we partition to 1 file per resource type. This gives fewer files for customers to download. 

When we have more than 100 MB of data we allow Glue to create as many files as needed per `resourceType`. With a DDB table that is several GB in size I've seen Glue makes a most of 19 files per `resourceType`. Each file is atleast 10 MB in size.

Reply:
I feel like this 100MB behavior is confusing for customers. I imagine that they'll try FWoA with test data and think that export returns one file per resource and only later when they run it in a prod-like environment they'll find out that it actually returns many files per resource.
Yes, the FHIR spec says that there can be multiple files per resource so it is compliant, but the 100MB threshold seems arbitrary. We could document this behavior but it sounds weird to me. 

Also, is there a way to set a maximum size for the S3 files that are written? That may be the ideal approach( idk if that's possible, I'm new to Spark)

Reply:
After further research and testing, I was able to get the code working with `repartition(1)` instead of `coalesce(1)`. I tested it by exporting 4GB of data from DDB and the code worked as expected. Further details about `coalesce` vs `repartition` [here](https://stackoverflow.com/a/56263274)

Reply:
Yep, setting a maximum size for S3 files written by Spark would be ideal. but unfortunately that's not possible. There are some hacky ways to do it by trying to approximate how much a file size would be given how many records you're writing to a file. However, that would cause the job to run slowly. 

This [link](https://stackoverflow.com/a/47229465) is about outputting the files to parquet, which is not exactly the same as json, but I believe the result will be the same.